R/modify_legend.R

```diff
@@ -208,3 +214,32 @@ modify_legend <- function(object, location="topright", legend_offset=0.3, draw=F
   object[['legend']][[paste0("legend.", legend.index)]] <- legend.config
   return(object)
 }
+
+#' get vector of legend arguments - overall or for each entry
+#' 
+#' @param overall logical indicating whether overall legend arguments should be returned. These arguments impact 
+#' the legend style (e.g. title, box.lwd, location, etc)
+#' @param indiv logical indiciating whether arguments applicable to each legend entry should be returned. These are 
+#' the arguments that apply to each individual entry within the legend (e.g. lty, lwd, pch, text.font, etc)
+#' @keywords internal
+get_legend_arg_names <- function(overall = FALSE, indiv = FALSE, names.args = names(formals(graphics::legend))){
```

Comment:
thinking of different things to do for the `names(formals...)` etc, because I think we want to have the handling of that be consistent for the package. 

This 
```r
formal_names('legend')
```
would be preferred, but you can't drop it in as the default because `formal_names` is a `gsplot:::` function. Perhaps use default of NULL which means all or `"all"` as default which means call `formal_names('legend')` within the function?